### PR TITLE
Fix decal crash #15190 (Stabilization request)

### PR DESCRIPTION
**Pull request by chanmosq**
**Exception request** -- This PR resolves a crash that occurs when updating a decal's material. As this is a common workflow with decals, I believe this is critical.

A separate [PR](https://github.com/o3de/o3de/pull/15719) for development created. 

## What does this PR do?

Fix decal crash #15190
- Changed how the image asset is verified to correctly catch the missing image error. Previously a broken image was considered valid and failed to be caught. 
- Changed Assert to Error so console can report missing image to the user and Editor can continue safely.

## How was this PR tested?

Tested manually by doing the following steps: 
Setup: 
1. Created a new material with a base color texture
2. Added the Decal component to the default ShaderBall
3. Added the material to the Decal component
4. Deleted the base color texture
5. Saved and closed Editor.
6. Relaunched Editor. 

Before this fix, the Editor would crash after loading a level. After the fix, the level loads and the "missing image" error is output to the console. 

## Exception Checklist

- [x] 1. Owning SIG requests the exception, by a Sig Representative (Chair/co-chair) adding the PR to the exception requests queue board.
- [ ] 2. (SIG-Content) Owning SIG comments on the PR explaining impact if this code does not go into stabilization.
- [ ] 3. Unrelated SIG representative (chair/cochair) approves the exception, adding a comment indicating approval and the SIG they represent.
- [x] 4. UX SIG representative (chair/cochair) approves the exception, adding a comment indicating approval and the SIG they represent.
- [x] 5. Testing SIG representative (chair/cochair) approves the exception, adding a comment indicating approval and the SIG they represent.
- [ ] (Must occur at some point before SIG-Release approval) Code has been approved for merging by maintainers
- [x] 6. SIG-Release Release Manager or co-Release Manager approves the exception, adding a comment indicating approval and the SIG they represent.
- [ ] 7. If all approvals are accepted, then the exception PR may be merged
- [ ] 8. Merged PR is verified successful via testing. If Merged PR verification testing failed, consult reps from step 3-5 and discussion occurs on fixing the failure vs backing out change
- [ ] 9. Exception is complete

### CODE REVIEW

**Review by invertednormal — APPROVED** [review]

**Review by rgba16f — APPROVED** [review]

File: Gems/Atom/Feature/Common/Code/Source/Decals/DecalTextureArrayFeatureProcessor.cpp [hunk 1]
```diff
@@ -32,14 +32,13 @@ namespace AZ
...
-                        const auto& assetId = imageBinding.GetId();
-                        if (assetId.IsValid())
+                        if (imageBinding && imageBinding.IsReady())
                         {
                             return imageBinding->GetImageDescriptor().m_size;
                         }
                     }
                 }
-                AZ_Assert(false, "GetSizeFromMaterial() unable to find an image in the given material.");
+                AZ_Error("DecalTextureArrayFeatureProcessor", false, "GetTextureSizeFromMaterialAsset() unable to find an image in the given material.")
```
Comment (Kadino) [inline Gems/Atom/Feature/Common/Code/Source/Decals/DecalTextureArrayFeatureProcessor.cpp:41]: May be helpful to output more information about which material / asset / file-path encountered this issue. Could even be appropriate to dump the contents of `GetPropertyValues()` for more info.

**Review by Kadino — APPROVED** [review]
Approved on behalf of SIG-Testing. Fixing this crash is an appropriately critical change.

**Comment by rainbj** [comment]
Sig-UI-UX approved

**Comment by tonybalandiuk** [comment]
SIG-Release Approves

**Comment by chanmosq** [comment]
The error message now outputs the source path of the material file. 
Editor_BjDHl0nhUD

**Review by invertednormal — APPROVED** [review]

**Comment by AMZN-alexpete** [comment]
Why does this need to go through the exception process? Aren't criticals and blockers allowed in without it till 4/21?

**Review by moudgils — APPROVED** [review]

**Comment by tonybalandiuk** [comment]
@AMZN-alexpete you are correct. I clarified with @chanmosq that this doesn't need an exception but forgot to put a note here. Sorry for the confusion. 

@chanmosq has this been tested.

**Comment by chanmosq** [comment]
Yes I ran the same manual steps listed above, and it passed the AR tests.

**Pull request merged**